pyproject.toml

```diff
@@ -25,3 +25,93 @@ extension-pkg-whitelist = [
 ]
 load-plugins = "pylint_fixme_info,pylint_pytest"
 good-names = "id"
+
+[tool.ruff]
+line-length = 88
+extend-exclude = [
+    "docs",
+    "__pycache__"
+]
+select = [
+    "E", "F",  # The rules built into Flake8
+    "D",  # Implement pydocstyle checking as well.
+    "I",  # isort checking
+    "PLC", "PLE", "PLR", "PLW",  # Pylint
+    # Additional stricter checking than previously enabled:
+    "A",  # Shadowing built-ins.
+    "W",  # PyCodeStyle warnings.
+    "N",  # Pep8 naming
+    "YTT",  # flake8-2020: Misuse of `sys.version` and `sys.version_info`
+    "ANN",  # Annotations
+    "S",  # Checks for common security issues
+    "BLE",  # Blind exception
+    "B",  # Opinionated bugbear linting
+    "C4",  # better comprehensions
+    "T10",  # Ensure we don't leave code that starts the debugger in
+    "ICN",  # Unconventional import aliases.
+    "Q",  # oh gawd there's gonna be a lot of this isn't there
```

Comment:
what's this one?

Reply:
Lol oops I forgot to update the comment. The comment was meant for when I tried the `T20` rules which complain about each usage of `print` (I was curious - it makes sense for a web API but not for what we're using here).

Q is for consistent quotations. Line 101 turns off sensing bad quotes in inline strings just because we have 2 single-quoted strings.

Related: I [made a feature request](https://github.com/charliermarsh/ruff/issues/1256) for a warning level in ruff so we can have things like that enabled as warnings while still allowing them to pass linting in CI.